### ISSUE TITLE
tags specified at the subnet level has no effect

### ISSUE THREAD

**Issue by vivgoyal-aws** [aws-ia/terraform-aws-vpc#77]
I am expecting that tags specified at each subnet level are applied to the specified subnets, merged with the tags specified at the VPC level.
The outcome is that only VPC level tags are applied. Subnet level tags are ignored.

**Comment by drewmullen** [comment]
hmm looks like subnets should get the tags... ill investigate today and get back to you

```
resource "aws_subnet" "private" {
  ...
  tags = merge(
    { Name = "${local.subnet_names[split("/", each.key)[0]]}-${split("/", each.key)[1]}" },
  module.tags.tags_aws)
}
```

**Comment by drewmullen** [comment]
I see. the issue is that we allow for a tags parameter within the var.subnet definition for each individual type. That is not accounted for at all in the resource definitions

**Comment by vivgoyal-aws** [comment]
Thanks for looking into it so quickly. I am building a solution where subnet level tags are very important

**Comment by vivgoyal-aws** [comment]
This issue is not fully fixed. If I have the tags in the `transit_gateway` subnet. 
```
...
    transit_gateway = {
      name_prefix             = "tgw"
      cidrs                   = ["10.50.11.0/28", "10.50.12.0/28"]
      connect_to_public_natgw = true
      #route_to_transit_gateway = "pl-"

      transit_gateway_id                              = "tgw-123"
      transit_gateway_default_route_table_association = false
      transit_gateway_default_route_table_propagation = false
      transit_gateway_appliance_mode_support          = "disable"
      transit_gateway_dns_support                     = "enable"

      tags = var.vpc_shared_svc_tgw_subnet_tags
    }
...
```
it breaks with following error
```
   on .terraform\modules\shared_services_vpc\main.tf line 210, in resource "awscc_ec2_route_table" "tgw":
│  207:   tags = concat(
│  208:     [{ "key" = "Name", "value" = "${local.subnet_names["transit_gateway"]}-${each.key}" }],
│  209:     module.tags.tags,
│  210:     try(module.subnet_tags[split("/", each.key)[0]].tags, {})
│  211:   )
│     ├────────────────
│     │ each.key is "us-east-1b"
│     │ module.subnet_tags is object with 3 attributes
│
│ Invalid value for "seqs" parameter: all arguments must be lists or tuples; got object.
```

Looks like issue is at ` try(module.subnet_tags[split("/", each.key)[0]].tags, {})`, it should be `try(module.subnet_tags["transit_gateway"].tags_aws, {})` instead.

**Comment by drewmullen** [comment]
Should be a list when !try also... I'll shoot up a fix tomorrow

Ty

**Comment by drewmullen** [comment]
2nd bug closed in #80
